File: topics/control-flow/application/PerformGuess.pas
function PerformGuess(num_guess, target: Integer): Boolean
var
    guess: Integer;
begin
    
    Write('Guess ', num_guess, ': ');
    Readln(guess);
    
    if target < guess then
    begin
        WriteLn('The number is less than ', guess);
    end
    else
    begin
        if target > guess then
            WriteLn('The number is larger than ', guess)
        else 
            WriteLn('Well done... the number was ', guess);
    end;
    
    result := target = guess;
end;

File: topics/control-flow/application/perform-guess.c
bool perform_guess(int num_guess, int target)
{
    int guess;
    
    printf("Guess %d: ", num_guess);
    scanf("%d", &guess);
    
    if (target < guess) 
    {
        printf("The number is less than %d\n", guess);
    }
    else
    {
        if (target > guess) 
            printf("The number is larger than %d\n", guess);
        else 
            printf("Well done... the number was %d\n", guess);
    }
    
    return target == guess;
}